Framework/CurveFitting/src/Functions/ComptonScatteringCountRate.cpp

```diff
@@ -233,31 +232,32 @@ void ComptonScatteringCountRate::updateCMatrixValues() const {
  * Cache workspace reference. Expects a MatrixWorkspace
  * @param matrix A shared_ptr to a Workspace
  * @param wsIndex A workspace index
- * @param startX Starting x-vaue (unused).
- * @param endX Ending x-vaue (unused).
+ * @param startX Starting x-value (unused).
+ * @param endX Ending x-value (unused).
  */
 void ComptonScatteringCountRate::setMatrixWorkspace(
     boost::shared_ptr<const API::MatrixWorkspace> matrix, size_t wsIndex,
     double startX, double endX) {
   CompositeFunction::setMatrixWorkspace(matrix, wsIndex, startX, endX);
 
-  const auto &values = matrix->readY(wsIndex);
-
+  this->matrix = matrix;
+  this->wsIndex = wsIndex;
```

Comment:
I wished to replace `m_errors` with the `HistogramE` equivalent so I could use `matrix->e(wsIndex)` to read in the error data. However, `m_errors` was defined in the header file and thus could not be initialised again. Saving the matrix was a workaround to allow this method to be used.